### ISSUE TITLE
Chef Provisioner: windows, lost escape for path

### ISSUE THREAD

**Issue by partamonov** [hashicorp/terraform#4039]
Chef provisioner:
os type: windows
ERROR: 
```aws_instance.cis (chef): Starting initial Chef-Client run...
aws_instance.cis (chef): [2015-11-24T09:11:47+00:00] FATAL: Cannot load configuration from C:cheffirst-boot.json```

Looks like problem is somewhere in `func (p *Provisioner) runCommand` as `func (p *Provisioner) runChefClientFunc` cmd variable looks correct (cmd /c chef-client -j "C:\\chef\\first-boot.json")

In Debug I see command in Chef Provisioner is fine: 
```2015/11/24 11:52:42 [DEBUG] vertex root, waiting for: provider.aws (close)
aws_instance.cis (chef): [2015-11-24T10:52:42+00:00] FATAL: Cannot load configuration from C:cheffirst-boot.json
2015/11/24 11:52:43 [DEBUG] terraform-provisioner-chef.exe: 2015/11/24 11:52:43 remote command exited with '0': cmd /c chef-client -j "C:\\chef\\first-boot.json" -E "development"
2015/11/24 11:52:43 [DEBUG] root: eval: *terraform.EvalIf```

**Comment by svanharmelen** [comment]
Fixed by PR #4041

**Comment by phinze** [comment]
Just landed #4041 - sorry for the trouble!
